### ISSUE TITLE
Allow selectively ignoring peer dep warnings

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#1125]
Some peer dep warnings don't prevent app from running such as some incompatible versions, whereas some such as missing peer deps usually would.

I would like to be able to silence a specific warning.

A message would say for example: "10 warnings have been explicitly silenced. Please see package.json pnpm.silencedWarnings."

# Implementation

The easiest way might be to copy the line from the log.

E.g.

```
"silencedErrors": [
  "deck.gl@4.1.6 requires a peer of luma.gl@^4.0.5 but none was installed."
]
```

Would be better in a yaml file or a js file so comments could be made about the issue that we are waiting on.

# Example

For example, here are my current warnings for a project that works fine:

```
WARN  deprecated supertest-as-promised@2.0.2: SuperTest 2.0+ supports promises natively; use that instead!
 WARN  deck.gl@4.1.6 requires a peer of luma.gl@^4.0.5 but none was installed.
 WARN  fixed-data-table@0.6.0 requires a peer of react@>=0.13.0 <0.15.0 || ^0.14.0-beta3 but version 15.6.2 was installed.
 WARN  fixed-data-table@0.6.0 requires a peer of react-dom@>=0.14.0 <0.15.0 || ^0.14.0-beta3 but version 15.6.2 was installed.
 WARN  graphql-bookshelf@0.0.8 requires a peer of graphql@^0.7.0 but version 0.13.0 was installed.
 WARN  graphql-server-koa: graphql-server-core@0.8.5 requires a peer of graphql@^0.9.0 || ^0.10.1 but version 0.13.0 was installed.
 WARN  graphql-server-koa@0.8.5 requires a peer of graphql@^0.9.0 || ^0.10.1 but version 0.13.0 was installed.
 WARN  graphql-tag@2.7.3 requires a peer of graphql@^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 but version 0.13.0 was installed.
 WARN  graphql-thinky: graphql-relay@0.4.4 requires a peer of graphql@^0.5.0 || ^0.6.0 || ^0.7.0 || ^0.8.0-b but version 0.13.0 was installed.
 WARN  graphql-thinky@0.4.0-rc-3 requires a peer of graphql@^0.8.0 but version 0.13.0 was installed.
 WARN  graphql-thinky@0.4.0-rc-3 requires a peer of thinky@^2.3.2 but version 2.2.4 was installed.
 WARN  react-addons-css-transition-group@0.14.7 requires a peer of react@^0.14.7 but version 15.6.2 was installed.
 WARN  react-apollo > enzyme-adapter-react-16: react-reconciler@0.7.0 requires a peer of react@^16.0.0 but version 15.6.2 was installed.
 WARN  react-apollo > enzyme-adapter-react-16: react-test-renderer@16.2.0 requires a peer of react@^16.0.0 but version 15.6.2 was installed.
 WARN  react-apollo: enzyme-adapter-react-16@1.1.1 requires a peer of enzyme@^3.0.0 but none was installed.
 WARN  react-apollo: enzyme-adapter-react-16@1.1.1 requires a peer of react@^16.0.0-0 but version 15.6.2 was installed.
 WARN  react-apollo: enzyme-adapter-react-16@1.1.1 requires a peer of react-dom@^16.0.0-0 but version 15.6.2 was installed.
 WARN  react-apollo: graphql-tag@2.7.3 requires a peer of graphql@^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 but version 0.13.0 was installed.
 WARN  react-bootstrap: react-overlays@0.5.4 requires a peer of react-dom@^0.14.0 but version 15.6.2 was installed.
 WARN  react-dnd@2.1.2 requires a peer of react@^0.14.0 but version 15.6.2 was installed.
 WARN  react-highcharts@5.0.6 requires a peer of react@~0.14 but version 15.6.2 was installed.
 WARN  react-highcharts@5.0.6 requires a peer of react-dom@~0.14 but version 15.6.2 was installed.
 WARN  react-leaflet@0.9.0 requires a peer of react@^0.14.0 but version 15.6.2 was installed.
 WARN  react-leaflet@0.9.0 requires a peer of react-dom@^0.14.0 but version 15.6.2 was installed.
 WARN  react-router-addons-routes@0.0.3 requires a peer of react-router@^4.0.0-2 but version 3.2.1 was installed.
 WARN  react-toolbox-themr@1.0.2 requires a peer of react-toolbox@^2.0.0 but version 2.0.0-beta.12 was installed.
```

@pgonzal

**Comment by zkochan** [comment]
Until we have such feature, you can use `pnpmfile.js` to avoid warnings about not satisfied ranges (not missing peer deps). Something like this: 

```js
module.exports = {
  hooks: {
    readPackages (pkg) {
      if (pkg.name === 'deck.gl') {
        pkg.peerDependencies['luma.gl'] = '*'
      }
      return pkg
    }
  }
}
```

**Comment by pgonzal** [comment]
On this same topic, we recently had reports about this warning:

![pnpm upgrade banner](https://user-images.githubusercontent.com/4673363/39145866-ee4b2bda-46e9-11e8-94a7-16cffe4ecc47.png)

In our repo, these instructions don't work because the globally installed pnpm is not used.  Instead the pnpm version is determined by a setting in `rush.json`.  

Can this warning be suppressed via `pnpmfile.js`?

**Comment by zkochan** [comment]
@pgonzal, you could run pnpm with the `--no-update-notifier` flag as @vjpr discovered in this issue: #1122

**Comment by pgonzal** [comment]
> @pgonzal, you could run pnpm with the --no-update-notifier flag as @vjpr discovered in this issue: #1122

Great, thanks very much!

**Comment by pgonzal** [comment]
[adding this to both issues for reference]

Here are examples of genuine problems that were being ignored by everybody because pnpm treats them as warnings:

> WARN @ms/plannerweb-ui: @ms/react-advanced-grid@8.1.5 requires a peer of moment@^2.22.0 but version 2.21.0 was installed.
> WARN @ms/searchux-sphome@36.0.3 requires a peer of react-router@>=2.6.1 but none was installed.
> WARN Cannot find file at C:\agent\3\_work\1\s\common\temp\node_modules\@rush-temp\react-bundle although it was listed by readdir
> WARN Cannot find file at C:\agent\3\_work\1\s\common\temp\node_modules\@rush-temp\react-dom-bundle although it was listed by readdir


Here are some example warnings about situations that are 100% intentional.  We will have no plans to fix them.  These false alarms lead people to never take any warnings seriously:

> WARN  using --no-lock I sure hope you know what you are doing
> WARN  webpack > watchpack > chokidar: Skipping failed optional dependency fsevents@1.2.0
> WARN  deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5

**Comment by pgonzal** [comment]
BTW I started to implement the workaround suggested here, but I realized we need `--warnings-as-errors` first (#1126).  Nobody is going to bother to suppress a warning in a situation where nobody has to pay any attention to the warnings.  :-)

**Comment by zkochan** [comment]
As far as I understand, this is only about peer dependencies currently. So maybe it can be a config called `--strict-peer`. As there are already similar flags: `--strict-ssl` and `--engine-strict`

**Comment by pgonzal** [comment]
Seems reasonable.  Generally our position on warnings is either (1) they must break a production build or else (2) we hide them so nobody sees them.  If we show warnings that don't break the build, it always turns into a huge pile of yellow text that everyone ignores, which undermines the whole model of warning people.

**Comment by muuvmuuv** [comment]
I'm also getting a lot of peer dependencies warnings in my project which works totally fine.

<details>
<summary>Show logs</summary>

```
 WARN  gatsby > @typescript-eslint/eslint-plugin > @typescript-eslint/experimental-utils > @typescript-eslint/typescript-estree: tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none was installed.
 WARN  gatsby > @typescript-eslint/eslint-plugin: tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none was installed.
 WARN  gatsby > @typescript-eslint/parser > @typescript-eslint/experimental-utils > @typescript-eslint/typescript-estree: tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none was installed.
 WARN  gatsby > @typescript-eslint/parser > @typescript-eslint/typescript-estree: tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none was installed.
 WARN  gatsby: react-hot-loader@4.12.19 requires a peer of @types/react@^15.0.0 || ^16.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-jest > babel-preset-jest: @babel/plugin-syntax-bigint@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-jest > babel-preset-jest: @babel/plugin-syntax-object-rest-spread@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-jest: babel-preset-jest@25.1.0 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb: babel-jest@25.1.0 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/plugin-proposal-class-properties: @babel/helper-create-class-features-plugin@7.8.3 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/plugin-proposal-class-properties@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/plugin-proposal-nullish-coalescing-operator: @babel/plugin-syntax-nullish-coalescing-operator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/plugin-proposal-nullish-coalescing-operator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/plugin-proposal-optional-chaining: @babel/plugin-syntax-optional-chaining@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/plugin-proposal-optional-chaining@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/plugin-syntax-dynamic-import@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/plugin-transform-runtime@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/plugin-transform-spread@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/helper-compilation-targets@7.8.4 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-async-generator-functions: @babel/plugin-syntax-async-generators@7.8.4 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-async-generator-functions@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-dynamic-import: @babel/plugin-syntax-dynamic-import@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-dynamic-import@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-json-strings: @babel/plugin-syntax-json-strings@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-json-strings@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-nullish-coalescing-operator: @babel/plugin-syntax-nullish-coalescing-operator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-nullish-coalescing-operator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-object-rest-spread: @babel/plugin-syntax-object-rest-spread@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-object-rest-spread@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-optional-catch-binding: @babel/plugin-syntax-optional-catch-binding@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-optional-catch-binding@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-optional-chaining: @babel/plugin-syntax-optional-chaining@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-optional-chaining@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-proposal-unicode-property-regex: @babel/helper-create-regexp-features-plugin@7.8.3 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-proposal-unicode-property-regex@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-async-generators@7.8.4 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-dynamic-import@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-json-strings@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-nullish-coalescing-operator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-object-rest-spread@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-optional-catch-binding@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-optional-chaining@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-syntax-top-level-await@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-arrow-functions@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-async-to-generator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-block-scoped-functions@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-block-scoping@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-classes@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-computed-properties@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-destructuring@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-transform-dotall-regex: @babel/helper-create-regexp-features-plugin@7.8.3 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-dotall-regex@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-duplicate-keys@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-exponentiation-operator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-for-of@7.8.4 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-function-name@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-literals@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-member-expression-literals@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-modules-amd@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-modules-commonjs@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-modules-systemjs@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-modules-umd@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-transform-named-capturing-groups-regex: @babel/helper-create-regexp-features-plugin@7.8.3 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-named-capturing-groups-regex@7.8.3 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-new-target@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-object-super@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-parameters@7.8.4 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-property-literals@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-regenerator@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-reserved-words@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-shorthand-properties@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-spread@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-sticky-regex@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-template-literals@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-typeof-symbol@7.8.4 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env > @babel/plugin-transform-unicode-regex: @babel/helper-create-regexp-features-plugin@7.8.3 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-env: @babel/plugin-transform-unicode-regex@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/preset-env@7.8.4 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-react: @babel/plugin-transform-react-display-name@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-react > @babel/plugin-transform-react-jsx: @babel/plugin-syntax-jsx@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-react: @babel/plugin-transform-react-jsx@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-react > @babel/plugin-transform-react-jsx-self: @babel/plugin-syntax-jsx@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-react: @babel/plugin-transform-react-jsx-self@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-react > @babel/plugin-transform-react-jsx-source: @babel/plugin-syntax-jsx@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby > @babel/preset-react: @babel/plugin-transform-react-jsx-source@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb > babel-preset-gatsby: @babel/preset-react@7.8.3 requires a peer of @babel/core@^7.0.0-0 but none was installed.
 WARN  gatsby-plugin-breadcrumb: babel-preset-gatsby@0.2.29 requires a peer of @babel/core@^7.0.0 but none was installed.
 WARN  gatsby-plugin-sass: sass-loader@7.3.1 requires a peer of webpack@^3.0.0 || ^4.0.0 but none was installed.
 WARN  gatsby-source-graphql > apollo-link: apollo-utilities@1.3.3 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql: apollo-link@1.2.13 requires a peer of graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > apollo-link-http > apollo-link: apollo-utilities@1.3.3 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > apollo-link-http: apollo-link@1.2.13 requires a peer of graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > apollo-link-http > apollo-link-http-common > apollo-link: apollo-utilities@1.3.3 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > apollo-link-http > apollo-link-http-common: apollo-link@1.2.13 requires a peer of graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > apollo-link-http: apollo-link-http-common@0.2.15 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql: apollo-link-http@1.5.16 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > graphql-tools-fork > apollo-link: apollo-utilities@1.3.3 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > graphql-tools-fork: apollo-link@1.2.13 requires a peer of graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > graphql-tools-fork > apollo-link-http-common > apollo-link: apollo-utilities@1.3.3 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > graphql-tools-fork > apollo-link-http-common: apollo-link@1.2.13 requires a peer of graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql > graphql-tools-fork: apollo-link-http-common@0.2.15 requires a peer of graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 but none was installed.
 WARN  gatsby-source-graphql: graphql-tools-fork@8.5.2 requires a peer of graphql@^14.2.0 but none was installed.
```

</detailas>

**Comment by crimx** [comment]
> Until we have such feature, you can use `pnpmfile.js` to avoid warnings about not satisfied ranges (not missing peer deps). Something like this:
> 
> ```js
> module.exports = {
>   hooks: {
>     readPackages (pkg) {
>       if (pkg.name === 'deck.gl') {
>         pkg.peerDependencies['luma.gl'] = '*'
>       }
>       return pkg
>     }
>   }
> }
> ```

Update the code according to the latest [docs](https://pnpm.io/pnpmfile). `pnpmfile.js` should be `.pnpmfile.cjs`. `readPackages` should be `readPackage`.

**Comment by zkochan** [comment]
Solved by #4212
